common/auth.go

```diff
@@ -15,6 +15,41 @@
 
 package common
 
-type AuthBackend interface {
-	
-}
\ No newline at end of file
+type Role interface {
+	Name() string
+}
+
+type role struct {
+	name string
+}
+
+// Represents the type of credential (e.g., certificate,
+// username-password, etc.
+type CredentialType string
+
+const (
+	CredentialUsernamePassword = "userPass"
+	CredentialNone             = "none"
+)
+
+// Container for various credentials. Currently containing Username/Password
+// but keys, certificates, etc. can be used in the future.
+type Credential struct {
+	Type     CredentialType
+	Username string
+	Password string
+}
+
+// MakeCredentialFromCliArgs takes all possible CLI
```

Comment:
Ok, please make sure to document the environment as the preferred way, so Caleb can modify the deployment scripts accordingly.